bodhi/models/models.py

```diff
@@ -1649,6 +1649,27 @@ def check_requirements(self, session, settings):
 
         return True, "All checks pass."
 
+    @property
+    def has_negative_karma(self):
+        """Check for negative karma, Returns True if the update has negative karma"""
+        for comment in self.comments:
+            if not comment.anonymous:
+                if comment.karma < 0:
+                    return True
+        return False
+
+    def diasble_autopush_for_critical_update(self, db):
+        """Disable Autopush if a critical update receives any negative karma"""
+        if not self.locked:
+            if self.status is UpdateStatus.testing:
+                if self.autokarma and self.critpath and self.has_negative_karma:
+                    log.info("Disabling Auto Push since the critical update has negative karma")
+                    self.autokarma = False
+                    db.commit()
+        else:
+            log.debug('%s locked. Ignoring karma thresholds.' % self.title)
+            raise LockedUpdateException
```

Comment:
Generally, it's good practice to instantiate Exception objects when raising them (though it's not strictly necessary).